### ISSUE TITLE
setup.py `install_requires` does not match requirements.txt

### ISSUE THREAD

**Issue by ottumm** [Yelp/elastalert#1014]
setup.py does not lock down dependencies in the same way as requirements.txt does. This caused me to run into https://github.com/Yelp/elastalert/issues/980 when deploying elastalert version 0.1.8, even though that version specifies a version of twilio that should not have caused this bug.

**Comment by ardentisys-rajshekhar** [comment]
@ottumm try installing on basis of requirements.txt then try with setup.py install. you should be able to install it.

**Comment by ottumm** [comment]
Yes, that's a workaround, but the point of this issue is that `pip install scrapy` will install a broken version because of issue #980. If `setup.py` locked down dependencies in the same way that `requirements.txt` does, this wouldn't be a problem.

**Comment by nrvnrvn** [comment]
@Qmando this is a simple task. Would you mind if I open a PR?

The only thing to choose is whether the source of truth will be in setup.py or in requirements.txt

In the former case requirements.txt will consist only `-e .` See [this article](https://caremad.io/posts/2013/07/setup-vs-requirement/).
In the latter case install_requires will parse requirements.txt. 

I've run into the issue recently and would like to help.
